modules/apps/object/object-test/src/testFunctional/tests/ObjectCustomLayouts.testcase

```diff
@@ -21,6 +21,136 @@ definition {
 		}
 	}
 
+	@description = "LPS-168228 - Verify it's possible to add many custom fields in the custom layout"
+	@priority = "4"
+	test CanAddManyFieldsOnLayout {
+		task ("Given an object with 15 fields is created") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 168228",
+				objectName = "CustomObject168228",
+				pluralLabelName = "Custom Objects 168228");
+
+			var n = "0";
+
+			while ("${n}" != "15") {
+				echo("${n}");
+
+				ObjectAdmin.addObjectFieldViaAPI(
+					fieldBusinessType = "Text",
+					fieldLabelName = "Custom Field${n}",
+					fieldName = "customField${n}",
+					fieldType = "String",
+					isRequired = "false",
+					objectName = "CustomObject168228");
+
+				var n = ${n} + 1;
+			}
+		}
+
+		task ("And given an layout is created in the object") {
+			ObjectAdmin.addObjectLayoutViaAPI(
+				layoutName = "Layout Name",
+				objectName = "CustomObject168228");
+
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 168228");
+
+			ObjectAdmin.goToLayoutsTab();
+
+			ObjectAdmin.selectKebabMenuOption(kebabOption = "View");
+
+			ObjectAdmin.markLayoutAsDefault();
+		}
+
+		task ("And given all the custom fields are added in the layout") {
+			ObjectAdmin.goToLayoutTabOnLayouts();
+
+			LexiconEntry.gotoAddNoSelectFrame();
+
+			Click(
+				key_type = "Fields",
+				locator1 = "ObjectAdmin#TAB_TYPE");
+
+			Type(
+				locator1 = "ObjectAdmin#TAB_LABEL_NAME",
+				value1 = "Field Tab");
+
+			Click(
+				key_tab = "Save",
+				locator1 = "Button#TAB_SUBMIT_BTN");
+
+			Click(locator1 = "ObjectAdmin#ADD_BLOCK_BUTTON");
+
+			Type(
+				locator1 = "ObjectAdmin#TAB_LABEL_NAME",
+				value1 = "Block 1");
+
+			Click(
+				key_tab = "Save",
+				locator1 = "Button#TAB_SUBMIT_BTN");
+
+			var z = "0";
+
+			while ("${z}" != "15") {
+				Click(locator1 = "ObjectAdmin#ADD_FIELD_ON_BLOCK");
+
+				Click(locator1 = "ObjectAdmin#LAYOUT_ENTRY_CHOOSE_OPTION");
+
+				Click(
+					key_optionName = "Custom Field${z}",
+					locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_OPTION");
+
+				Click(
+					key_columnsNumber = "1",
+					locator1 = "ObjectAdmin#LAYOUT_ADD_FIELD_COLUMN_NUMBER");
+
+				Click(
+					key_tab = "Save",
+					locator1 = "Button#TAB_SUBMIT_BTN");
+
+				var z = ${z} + 1;
+			}
+
+			Button.clickSave();
+		}
```

Comment:
There are different macros for adding **Custom Layout** on **Custom Object**. Search for: **ObjectAdmin.addTabFieldsOnLayout**.